src/orion/core/worker/experiment.py

```diff
@@ -257,9 +260,30 @@ def reserve_trial(self, score_handle=None):
             selected_trial = self.reserve_trial(score_handle=score_handle)
         else:
             selected_trial = Trial(**selected_trial_dict)
+            TrialMonitor(self, selected_trial.id).start()
 
         return selected_trial
 
+    def fix_lost_trials(self):
+        """Find lost trials and set them to interrupted.
+
+        A lost trial is defined as a trial whose heartbeat as not been updated since two times
+        the wait time for monitoring. This usually means that the trial is stalling or has been
+        interrupted in some way without its status being changed. This functions finds such
+        trials and set them as interrupted so they can be launched again.
+
+        """
+        query = {'experiment': self._id, 'status': {'$in': ['reserved']}}
```

Comment:
```suggestion
        query = {'experiment': self._id, 'status': 'reserved'}
```

Reply:
I would add `'heartbeat': {'$lte': datetime.datetime.utcnow() - datetime.timedelta(seconds=60 * 2)}` to limit the number of trials loaded and then you can get rid of the `if` in the `for` loop.

Please add a `TODO` comment to make sure we don't forget to modify this part when we make the heartbeat configurable.

Also, you'll need to add the `$lte` command here: https://github.com/Epistimio/orion/blob/master/src/orion/core/io/database/ephemeraldb.py#L281